src/Lexer.php

```diff
@@ -313,8 +402,16 @@ protected function lexRawData($tag)
         $text = substr($this->code, $this->cursor, $match[0][1] - $this->cursor);
         $this->moveCursor($text.$match[0][0]);
 
-        if (false !== strpos($match[1][0], $this->options['whitespace_trim'])) {
-            $text = rtrim($text);
+        // trim?
+        if (isset($match[1][0])) {
+            if ($this->options['whitespace_trim'] === $match[1][0]) {
+                // whitespace_trim detected ({%-, {{- or {#-)
+                $text = rtrim($text);
+            } else {
+                // whitespace_line_trim detected ({%~, {{~ or {#~)
+                // don't trim \r and \n
+                $text = rtrim($text, " \t\0\x0B");
```

Comment:
I think this should consume one `\r\n` if they are immediately behind the opening brackets. That allows 
```
{{~
```
not to be an NOOP and do something that can be useful.
E.g.
```
abcd

{{~
```
Equivalent to:
```
abcd
{{
```